azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/model/WebAppRuntime.java

```diff
@@ -0,0 +1,139 @@
+        if (this.isDocker()) {
+            return 0;
+        }
+        final String v = this.getJavaVersionNumber();
+        return Integer.parseInt(StringUtils.startsWithIgnoreCase(v, "1.") ? String.valueOf(v.charAt(2)) : v);
+    }
+
+    default WebContainer getWebContainer() {
+        if (this.isDocker()) {
+            return WebContainer.fromString("docker");
+        }
+        return WebContainer.fromString(String.format("%s %s", getContainerName().toLowerCase(), getContainerVersionNumber()));
+    }
+
+    default JavaVersion getJavaVersion() {
+        if (this.isDocker()) {
+            return JavaVersion.fromString("docker");
+        }
+        return JavaVersion.fromString(this.getJavaVersionNumber());
+    }
+
+    @Override
+    default boolean isMinorVersion() {
+        if (this.isDocker()) {
+            return false;
+        }
+        final Pattern EXCLUDE_PATTERN = Pattern.compile("\\..*\\.");
+        return EXCLUDE_PATTERN.matcher(this.getJavaVersionNumber()).matches() || EXCLUDE_PATTERN.matcher(this.getContainerVersionNumber()).matches();
+    }
+
+    default boolean isJavaSE() {
+        return StringUtils.startsWithIgnoreCase(this.getContainerName(), "java");
+    }
+
+    default boolean isJBoss() {
+        return StringUtils.startsWithIgnoreCase(this.getContainerName(), "jboss");
+    }
+
+    default boolean isTomcat() {
+        return StringUtils.startsWithIgnoreCase(this.getContainerName(), "tomcat");
+    }
+
+    default String getJavaVersionUserText() {
+        if (this.isDocker()) {
+            return "";
+        }
+        if (StringUtils.equalsAnyIgnoreCase(getJavaVersionNumber(), "1.8", "8")) {
+            return "Java 8";
+        }
+        return String.format("Java %s", getJavaVersionNumber());
+    }
+
+    default String getContainerUserText() {
+        if (this.isDocker()) {
+            return "Docker";
+        }
+        return String.format("%s %s", StringUtils.capitalize(getContainerName().toLowerCase()), getContainerVersionNumber());
+    }
+
+    default List<PricingTier> getPricingTiers() {
+        return getPricingTiers(this.getOperatingSystem(), this.getContainerUserText());
+    }
+
+    static List<WebAppRuntime> getMajorRuntimes() {
+        return Stream.concat(
+                Stream.of(WebAppDockerRuntime.INSTANCE), Stream.concat(
+                    WebAppWindowsRuntime.getMajorRuntimes().stream(),
+                    WebAppLinuxRuntime.getMajorRuntimes().stream()))
+            .collect(Collectors.toList());
+    }
+
+    static List<WebAppRuntime> getAllRuntimes() {
+        return Stream.concat(
+                Stream.of(WebAppDockerRuntime.INSTANCE), Stream.concat(
+                    WebAppWindowsRuntime.getAllRuntimes().stream(),
+                    WebAppLinuxRuntime.getAllRuntimes().stream()))
+            .collect(Collectors.toList());
+    }
+
+    static List<PricingTier> getPricingTiers(OperatingSystem os, String containerUserText) {
```

Comment:
why not use Runtime here instead of containerUserText

Reply:
because the caller doesn't specify `javaVersion`.